control-plane/subcommand/gateway-cleanup/command.go

```diff
@@ -192,3 +239,89 @@ func exponentialBackoffWithMaxIntervalAndTime() *backoff.ExponentialBackOff {
 	backoff.Reset()
 	return backoff
 }
+
+// loadGatewayConfigs reads and loads the configs from `/consul/config/config.yaml`, if this file does not exist nothing is done.
+func (c *Command) loadGatewayConfigs() error {
+	file, err := os.Open(c.flagGatewayConfigLocation)
+	if err != nil {
+		if os.IsNotExist(err) {
+			c.UI.Warn(fmt.Sprintf("gateway configuration file not found, skipping gateway configuration, filename: %s", c.flagGatewayConfigLocation))
+			return nil
+		}
+		c.UI.Error(fmt.Sprintf("Error opening gateway configuration file %s: %s", c.flagGatewayConfigLocation, err))
+		return err
+	}
+
+	config, err := io.ReadAll(file)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading gateway configuration file %s: %s", c.flagGatewayConfigLocation, err))
+		return err
+	}
+
+	err = k8syaml.Unmarshal(config, &c.gatewayConfig)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error decoding gateway config file: %s", err))
+		return err
+	}
+
+	if err := file.Close(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Command) deleteV2GatewayClassAndClassConfigs() error {
+	for _, gcc := range c.gatewayConfig.GatewayClassConfigs {
+
+		// find the class config and mark it for deletion first so that we
+		// can do an early return if the gateway class isn't found
+		config := &v2beta1.GatewayClassConfig{}
+		err := c.k8sClient.Get(context.Background(), types.NamespacedName{Name: gcc.Name, Namespace: gcc.Namespace}, config)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				// no gateway class config, just ignore and return
+				return nil
+			}
+			return err
+		}
+
+		// ignore any returned errors
+		_ = c.k8sClient.Delete(context.Background(), config)
+
+		// find the gateway class
+		gatewayClass := &v2beta1.GatewayClass{}
+		//TODO is the correct name for the gatewayclass?
+		err = c.k8sClient.Get(context.Background(), types.NamespacedName{Name: gcc.Name, Namespace: gcc.Namespace}, gatewayClass)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				// no gateway class, just ignore and return
+				return nil
+			}
+			return err
+		}
+
+		// ignore any returned errors
+		_ = c.k8sClient.Delete(context.Background(), gatewayClass)
+
+		// make sure they're gone
+		if err := backoff.Retry(func() error {
```

Comment:
why do we do this twice? any reason to not just do the entire delete in the retry block to begin with?